libs/framework-core/source/ftrack_framework_core/client/__init__.py

```diff
@@ -180,9 +180,7 @@ def tool_config_options(self):
         return self._tool_config_options
 
     def __init__(
-        self,
-        event_manager,
-        registry,
+        self, event_manager, registry, run_in_main_thread_wrapper=None
```

Comment:
Consider if we should always pass a "run_in_thread_wrapper" argument. And then just have a default that you can pass in when instancing the client:

`client.Client(..., default_thread_wrapper)`

This would simplify this class implementation and allow it to assume that it should always execute code in the self.default_thread_wrapper

Reply:
Agreed, but in any case, I think we should always allow the user to start it as None, because when in complete standalone mode without any UI, there is no reason to need to execute the methods in the main thread, so this way the user can decide to let them work in a separated thread.

Reply:
Perhaps we could do something as simple as this in the def __init__() of the Client/Host or wherever we use the run_in_main_thread_wrapper:

```python
if run_in_main_thread_wrapper:
   self.run_in_main_thread_wrapper = run_in_main_thread_wrapper
else:
   # "Pseudo code", not sure if this is exactly what you wanted.
   # Consider replacing the lambda with our own "util.call_directly".
   self.if run_in_main_thread_wrapper = lambda inner_function, *args, **kwargs: inner_function(*args, **kwargs)
```

The point of this exerice was just to allow class methods and decorqtaors to always call self.run_in_main_thread_wrapper and not check if it exists or not.

Reply:
Done, add suggested changes in a utility named call_directly. Also current run_in_main_thread decorator renamed to delegate_to_main_thread_wrapper and  add a run_in_main_thread for python standalone usage.